src/reflection.ts

```diff
@@ -842,10 +857,34 @@ export class Parser {
 
     const parseDiscriminator = this.readScalarLambda(discriminatorField, typeIndex, readDefaults);
 
-    const unionDeserializers = this.createUnionDeserializers(fieldType, readDefaults);
+    const fieldOffset = field.offset();
+
+    const unionDeserializers = new Map<number, (table: Table) => Record<string, any> | undefined>();
+
+    for (const [enumVal, type] of this.getUnionTypes(fieldType)) {
+      const specificTypeIndex = type.index();
+      if (specificTypeIndex < 0) {
+        continue;
+      }
+
+      const typeDeserializer = this.toObjectLambda(specificTypeIndex, readDefaults);
+      const deserializer = (table: Table) => {
+        const offsetToOffset = table.offset + table.bb.__offset(table.offset, fieldOffset);
+        if (offsetToOffset === table.offset) {
+          return undefined;
```

Comment:
I seem to have preferred to use `null` to indicate an unpopulated value for other field types, can we keep that consistent?

Reply:
Its not entirely clear. I would agree that there are some code paths that return null, but others seem to say they can return `| undefined`.

https://github.com/jkuszmaul/flatbuffers_reflection/blob/main/src/reflection.ts#L289
https://github.com/jkuszmaul/flatbuffers_reflection/blob/main/src/reflection.ts#L253
https://github.com/jkuszmaul/flatbuffers_reflection/blob/main/src/reflection.ts#L750

IMO, `undefined` is more idiomatic and natural. Some libraries like using `null` but I've always found it awkward. But it is your call so if you want `null` I can update the various types so it is clear what is expected.

Reply:
I'll be honest I have no idea why I made the type annotations on those indicate returning `undefined` rather than `null`....

In terms of stylistic choices I don't have a strong preference, although I think I have a slight preference towards `null` for this, following the logic in the first sentence of [undefined values have not been defined yet, whereas null values indicate intentional absence of a value.](https://dev.to/typescripttv/what-is-the-difference-between-null-and-undefined-5h76#:~:text=The%20convention%20in%20TypeScript%20is,intentional%20absence%20of%20a%20value.) (and I would argue in these cases the absence of a value is "intentional"). However I also see things that just generally discourage the use of `null` entirely and I don't mind adhering to common conventions.

We should definitely be consistent, though.

Reply:
That link is kind of meh - it is true that you could use it semantically like that - but in practice no one cares. `undefined` means both and that is sufficient.

Reply:
Want me to do that in this PR or follow-up?

Reply:
per separate conversation, any further changes will be separate.